docs/okeps/okep-5375-trust-zones.md

```diff
@@ -0,0 +1,350 @@
+accordingly.
+
+When `transport_zones` are set, the chassis will only create tunnel Interfaces
+for other chassis that also belong to at least one of the listed zones.
+
+When `transport_zones` are not set, tunnel Interfaces are created for all other
+chassis. But since peer chassis with `transport_zones` set will not set up
+tunnels to these implicitly marked chassis, no tunnels between marked and
+unmarked chassis will be created either.
+
+### OVN-Kubernetes
+
+Instead of using OVN Transport Zones directly, OVN-Kubernetes will maintain the
+list of remote `Chassis` in SBDB so that only chassis that share a trust zone
+with the local chassis will have a corresponding `Chassis` record in SBDB.
+Since there won't be `Chassis` records for chassis from disjoint trust zones,
+no tunnels will be created.
+
+To guarantee that a rogue chassis won't be able to change its `Node` annotation
+to expose the chassis to illegal trust zones, an admission hook should be
+installed to deny such updates. (Note: this is already the case.)
+
+### API Details
+
+#### DPUs
+
+The proposed solution should work in the presence of a DPU chassis, with and
+without a Host chassis. DPU chassis don't have a separate `Node` object, so
+they may have to use its Host `Node` object to communicate about trust zones.
+
+There may be more than one DPU chassis per Host. The proposed solution
+should support specifying trust zones for DPU chassis that are different from
+other chassis that belong to the same `Node` (whether on-DPU or Host).
+
+(In a similar vein, the proposed API format should be flexible enough for
+multiple virtual OVN chassis running on the same Host  or per-VTEP trust
+zones.)
+
+#### Node annotation
+
+To serve all possible permutations of chassis (DPU vs Host; numbers of each
+type; multiple VTEPs), a new `Node` object annotation of map type will be
+added.
+
+Each key in the map will represent a particular endpoint entity. In the scope
+of this OKEP, "endpoint entity" refers to an OVN chassis, but its definition
+may later be expanded to e.g. support per VTEP trust zones. For OVN chassis,
+`Chassis` `system-id` will be used as the key.
+
+Each value of the map will be a list of trust zones that the endpoint entity
+(chassis) belongs to.
+
+A default key `__default` will be used to represent the default trust zones for
+endpoints that do not have a specific key set.
+
+---
+
+For example, if all OVN chassis on a `Node` are part of the same trust zones,
+one can set the annotation like this:
+
+```yaml
+apiVersion: v1
+kind: Node
+metadata:
+  annotations:
+    k8s.ovn.org/trust-zones: '{"__default": ["tz1","tz2","tz3"]}'
+```
+
+This setting will serve scenarios with a single OVN chassis on the `Node`
+(either on-DPU or on-Host); or with multiple OVN chassis that all belong to the
+same trust zones.
+
+---
+
+If multiple OVN chassis are present on a `Node`, and they belong to different
```

Comment:
Reading through the code and trying to implement it (and talking to Nithya R), I now realize that OVN-K is perhaps not ready for multi-chassis Nodes, yet, so the initial implementation may assume node-chassis-id is the target key for now. A separate initiative may be needed to support multiple chassis per single Node. But the annotation syntax will be ready to support it when ready.